--- a/mt_wifi/embedded/common/cmm_cfg.c
+++ b/mt_wifi/embedded/common/cmm_cfg.c
@@ -13352,7 +13352,7 @@ INT set_ack_timeout_mode_byband(
 	RTMP_CHIP_OP *chip_ops = hc_get_chip_ops(pAd->hdev_ctrl);
 	INT ret = TRUE;
 
-	if ((timeout > MAX_ACK_TIMEOUT)) {
+	if (((timeout & MAX_ACK_TIMEOUT) > MAX_ACK_TIMEOUT)) {
 		MTWF_DBG(pAd, DBG_CAT_ALL, DBG_SUBCAT_ALL, DBG_LVL_ERROR,
 			"CTS/ACK Timeout Range should between [0xFFFF:0]!!\n");
 		return FALSE;
